Tools/MonoGame.Content.Builder/MonoGame.Content.Builder.targets

```diff
@@ -0,0 +1,302 @@
+    </PropertyGroup>
+
+  </Target>
+
+  <!--
+    Converts MonoGameContentReference items to ContentReference items, deriving the necessary metadata.
+
+    Outputs:
+      - ContentReference: references to .mgcb files that can be built with MGCB
+        - FullDir: the absolute path of the folder containing the .mgcb file
+        - ContentDir: the relative path of the resource folder to contain the content files
+        - ContentOutputDir: the absolute path of the bin folder containing final built content
+        - ContentIntermediateOutputDir: the absolute path of the obj folder containing intermediate content
+
+    Example:
+      - Given the following file setup:
+        - C:\Game\Game.Shared\Content.mgcb
+        - C:\Game\Game.DesktopGL\Game.DesktopGL.csproj
+          - MonoGameContentReference: ..\Content\Content.mgcb
+      - Output:
+        - ContentReference
+          - FullDir: C:/Game/Game.Shared/
+          - ContentDir: Game.Shared/
+          - ContentOutputDir: C:/Game/Game.Shared/bin/DesktopGL/Content
+          - ContentIntermediateOutputDir: C:/Game/Game.Shared/obj/DesktopGL/Content
+  -->
+  <Target Name="CollectContentReferences">
+
+    <ItemGroup Condition="'$(EnableMGCBItems)' == 'true'">
+      <MonoGameContentReference Include="**/*.mgcb" />
+    </ItemGroup>
+
+    <ItemGroup>
+
+      <!-- Start with existing metadata. -->
+      <ContentReference Include="@(MonoGameContentReference)">
+        <Link>%(MonoGameContentReference.Link)</Link>
+        <FullDir>%(MonoGameContentReference.RootDir)%(MonoGameContentReference.Directory)</FullDir>
+        <ContentFolder>%(MonoGameContentReference.ContentFolder)</ContentFolder>
+        <OutputFolder>%(MonoGameContentReference.Filename)</OutputFolder>
+      </ContentReference>
+
+      <!--
+        Process intermediate metadata.
+        Switch all back-slashes to forward-slashes so the MGCB command doesn't think it's trying to escape characters or quotes.
+        ContentFolder will be the name of the containing folder (using the Link if it exists) so the directory structure of the included content mimics that of the source content.
+      -->
+      <ContentReference>
+        <FullDir>$([System.String]::Copy(%(FullDir)).Replace('\','/'))</FullDir>
+        <ContentFolder Condition="'%(ContentFolder)' == '' AND '%(Link)' != ''">$([System.IO.Path]::GetDirectoryName(%(Link)))</ContentFolder>
+        <ContentFolder Condition="'%(ContentFolder)' == '' AND '%(Link)' == '' AND '%(RelativeDir)' != ''">$([System.IO.Path]::GetFileName($([System.IO.Path]::GetDirectoryName(%(RelativeDir)))))</ContentFolder>
+      </ContentReference>
+
+      <!-- Assemble final metadata. -->
+      <ContentReference>
+        <ContentDir>%(ContentFolder)/</ContentDir>
+        <ContentOutputDir>%(FullDir)bin/$(MonoGamePlatform)/%(OutputFolder)</ContentOutputDir>
+        <ContentIntermediateOutputDir>%(FullDir)obj/$(MonoGamePlatform)/%(OutputFolder)</ContentIntermediateOutputDir>
```

Comment:
I'm good with merging this PR as is, but we should look into fixing the directories that are used. The built content should also be put in the intermediate output folder, instead of bin. I'd suggest something like the following:
- `%(IntermediateOutputPath)/mgcontent/$(MonoGamePlatform)/$(OutputFolder)` for the MGCB intermediate folder.
- `%(IntermediateOutputPath)/xnb/$(MonoGamePlatform)/$(OutputFolder)` for the XNB's.

Reply:
Fair point. I wanted to try to keep this a straight translation, with fixes separate. Also, do we ever plan to use the bin/obj values specified in the mgcb?